### PR TITLE
doc: fix scripts and regenerate

### PR DESCRIPTION
**Pull request by tjdevries**
OK, fixed up some script issues that I've had with documentation generation.

Also, I fixed many missing pieces of documentation (and will encourage people to do the documentation in future PRs).

I have a few more things to fix in the scripts and then the generated output. I need this so that I can actually generate the help output for some other PRs :)

Closes: #12163

^^ That is getting closed because this takes the incremental fixes from there and ports them here. I will, in a separate PR in the future, create a tree-sitter parser / generator for the docs (at least for Lua, since doxygen is good for C).

### CODE REVIEW

File: runtime/doc/lsp.txt [hunk 1]
```diff
@@ -946,59 +966,61 @@ apply_text_edits({text_edits}, {bufnr})
...
+buf_diagnostics_count({kind})           *vim.lsp.util.buf_diagnostics_count()*
+                Returns the number of diagnostics of given kind for current
+                buffer.
+
+                Useful for showing diagnostic counts in statusline. eg:
 >
+
     function! LspStatus() abort
         let sl = ''
         if luaeval('vim.lsp.buf.server_ready()')
```
Comment (tjdevries) [inline runtime/doc/lsp.txt:984]: TODO: This should get changed to something that doesn't sent server notifications.

**Comment by h-michael** [comment]
@tjdevries 
If I remember correctly, I think the results of the execution changed slightly depending on the version of Doxygen.
I think that was one of the reasons why justinmk was doing the doc updates.
What version of Doxygen did you use?

**Comment by tjdevries** [comment]
```
➜ doxygen --version                                                    
1.8.19 (cd581388f3d013c501e3cefbaf3e81cf93d46fcb)
```

I have not tested with other versions, but I think I may have fixed some problems that occurred due to different versions. (there were some problems with list items).

I also modified the formatting for better `<pre></pre>` sections, which is why you see those diffs. That is for aesthetics, not due to doxygen version :)

File: runtime/doc/api.txt [hunk 2]
```diff
@@ -1757,18 +1788,18 @@ nvim_buf_get_extmarks({buffer}, {ns_id}, {start}, {end}, {opts})
...
-                  -- Create new extmark at line 1, column 1.
-                  local m1  = a.nvim_buf_set_extmark(0, ns, 0, 0, 0, {})
-                  -- Create new extmark at line 3, column 1.
-                  local m2  = a.nvim_buf_set_extmark(0, ns, 0, 2, 0, {})
-                  -- Get extmarks only from line 3.
-                  local ms  = a.nvim_buf_get_extmarks(0, ns, {2,0}, {2,0}, {})
-                  -- Get all marks in this buffer + namespace.
-                  local all = a.nvim_buf_get_extmarks(0, ns, 0, -1, {})
-                  print(vim.inspect(ms))
+    local a   = vim.api
```
Comment (justinmk) [inline runtime/doc/api.txt:1791]: @tjdevries this indentation seems like a regression, was it intentional for some reason?

**Pull request merged**